Walkthrough Guide/09 Mobile Network Services/README.md

```diff
@@ -25,14 +25,25 @@ It requires us to define our REST API calls as a C# Interface which is then used
 
 
 #### Security 
-Because we’re using Azure API Management, we have the ability to restrict access to our APIs through the use of API Keys. With a unique API key, we’re able to confirm that this app is allowed access to our services. We’ll want to ensure we add the API key to all our requests and Refit makes this super easy! We just need to add the _Headers_ attribute to our interface. Here we grab our API key from the constants class. 
+Because we’re using Azure API Management, we have the ability to restrict access to our APIs through the use of API Keys. With a unique API key, we’re able to confirm that this app is allowed access to our services. We’ll want to ensure we add the API key to all our requests and Refit makes this super easy! We just need to set the `Ocp-Apim-Subscription-Key` header in every request. Here we grab our API key from the constants class and add it to the default headers for the created services. 
 
 
  ```cs
-[Headers(Helpers.Constants.ApiManagementKey)]`
-public interface IJobServiceAPI`
+public abstract class BaseAPIService
 {
+    protected TService GetManagedApiService<TService>()
+    {
+        // Here we set up the stuff that is the same in each http request for our api.
+        // In this case it's the base url and the api management key header.
+        var client = new HttpClient
```

Comment:
Can we possibly reuse the HTTP Client if we've already created one? Currently we'd be creating a new client for each request.

Reply:
I think you can. I thought about it, but didn't want to introduce side effects regarding DNS cache stuff. If you wan't just add `Lazy<HttpClient>` :)

Reply:
I believe we we're originally new-ing up an instance of HttpClient for every request so your change isn't breaking the existing solution but rather brought to my attention to something we could improve. 

I'm guessing the DNS caching side effect you mention is related to the HttpClient instance holding DNS info for 2 minutes? If thats the case then we should* be safe given we use API Management in production and the DNS is locked down and not changing but with that said, I think we can make this all work nicely...

**One vs multiple clients**
To quote the [docs on HttpClient](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.-ctor?view=netframework-4.7.2): 

> HttpClient is intended to be instantiated once and re-used throughout the life of an application. Instantiating an HttpClient class for every request will exhaust the number of sockets available under heavy loads. This will result in SocketException errors. 

Sockets can be held for up to 4 minutes per request so this might actually be causing our network stability issues when we send too many requests. I'd love @robinmanuelthiel to share his thoughts on this as well. 

**Possible solution to DNS Cache with 1 instance**
We could use set the `DnsRefreshTimeout ` property of [ServicePointManager ](https://docs.microsoft.com/en-us/dotnet/api/system.net.servicepointmanager?view=netframework-4.7.2) class to set our own DNS refresh timeout to something more sensible (if we REALLY needed to). This could help mitigate any issues with DNS caching.